tensorflow/core/common_runtime/dml/dml_device_context.cc

```diff
@@ -35,19 +35,13 @@ void DMLDeviceContext::CopyCPUTensorToDevice(const Tensor* cpu_tensor,
 
   const void* src_data = DMAHelper::base(cpu_tensor);
 
-  D3D12BufferRegion dst = dml_util::CreateBufferForTensor(
-      static_cast<DmlDevice*>(device), *device_tensor);
-  ID3D12Resource* dst_data = dst.Resource();
-  const uint64_t dst_offset = dst.Offset();
-  const auto dst_state =
-      D3D12_RESOURCE_STATE_UNORDERED_ACCESS;  // GPU resources are always kept
-                                              // in UAV state
+  D3D12BufferRegion dst = CreateBufferForTensor(*device_tensor);
```

Comment:
It might actually be worthwhile to rename this to "GetBufferForTensor". It used to be called "Create" because the logical lifetime of the buffers was determined by the returned D3D12BufferRegion. As far as the caller was concerned, CreateBufferForTensor really did create a fresh placed resource that disappeared once the D3D12BufferRegion went away (the fact that they were really cached in a pool was an implementation detail). If the new lifetime semantics are intended to be different, a rename of this function might be warranted.

Reply:
I completely agree, the naming is confusing. I tried to avoid renaming helpers to reduce mental burden of doing this code review (tons of refactoring already) but I'm fine with doing the change here.